### ISSUE TITLE
FAQ: Does the RegEx support absolute URL checks like `^/file-on-root.txt`

### ISSUE THREAD

**Issue by porg** [BoldGrid/w3-total-cache#682]
https://github.com/BoldGrid/w3-total-cache/wiki/FAQ:-Usage#which-textareas-for-file-entries-support-regular-expressions

Does not address my question:
- Is a RegEx like `^/.*txt` matching all text files, but only if directly on the root level? (Due to the caret `^` as the start position marker)
- Or `^/.+/.jpe*g` only matching JPEG files on level 2?

**Comment by mavas84** [comment]
Hello @porg 

Thank you for reaching out and answering your questions:

-

>  Is a RegEx like ^/.*txt matching all text files, but only if directly on the root level? (Due to the caret ^ as the start position marker)
The answer is yes!

> Or ^/.+/.jpe*g only matching JPEG files on level 2?

no because .+ matches / char too

Thanks!

**Comment by porg** [comment]
1. Let me rephrase the overarching question: Are the RegEx positional markers `^` for line start and `&` for line end regarded in order to distinguish between a file anywhere or a file at a particular depth in relation to root?

2. Yeah and thanks for also addressing the detail question.
- That although the slash character is special in filesystem / URL context, it is included in `.` (any character)
  - So you perform no "web/filesystem-specific" flavor of RegEx, but normal RegEx.
- My [sample data on Regex101.com](https://regex101.com/r/9JKrr4/1) :
    ```
    /bird.jpg
    /doc.pdf
    /pets/cat.jpeg
    /pets/dog.jpg
    /pets/list.txt
    /zoo/has/animals.jpg
    ```
  - `[^/]+\.jpe*g` to match [JPEG files everywhere](https://regex101.com/r/9JKrr4/2) 
  - `/[^/]+/[^/]+\.jpe*g` to match [JPEG files with at least one directory above](https://regex101.com/r/9JKrr4/3)
  - `^/[^/]+/[^/]+\.jpe*g` to match [JPEG files within exactly one directory above in root](https://regex101.com/r/9JKrr4/4)
  - In order for this to work, positional markers would need to be supported. Are they?

**Comment by cssjoe** [comment]
The entries accept PCRE regex which is compared to each file path.   A good tool to test regex is https://regex101.com/

**Comment by porg** [comment]
I had asked very specifically (where in the URL do positional markers match?) and you give me a very general purpose answer. Please read again and answer carefully. Thanks!
